src/components/networked-hand-controls.js

```diff
@@ -0,0 +1,356 @@
+  
+      this.loader.load(this.data.handModelURL || handmodelUrl, 
+                       gltf => {
+        const newMesh = gltf.scene.children[0];
+        // const mesh = gltf.scene.children[0];
+        const handModelOrientation = this.data.hand === 'left' ? Math.PI / 2 : -Math.PI / 2;
+        newMesh.mixer = new THREE.AnimationMixer(newMesh);
+        
+        this.clips = gltf.animations;
+        this.clips.forEach((clip, clipIndex) => {
+          this.clipNameToClip[clip.name] = clip;
+        })      
+        
+        this.el.setObject3D('mesh', newMesh);
+  
+        const handMaterial = newMesh.children[1].material;
+        handMaterial.color = new THREE.Color(this.data.handColor);
+        newMesh.position.set(0, 0, 0);
+        newMesh.rotation.set(0, 0, handModelOrientation);
+  
+        if (this.local) {
+          const controlConfiguration = {
+            hand: this.data.hand,
+            model: false,
+          }        
+          this.el.setAttribute('magicleap-controls', controlConfiguration);
+          this.el.setAttribute('vive-controls', controlConfiguration);
+          this.el.setAttribute('oculus-touch-controls', controlConfiguration);
+          this.el.setAttribute('windows-motion-controls', controlConfiguration);
+          this.el.setAttribute('hp-mixed-reality-controls', controlConfiguration);
+          this.isViveController();
+        }
+  
+      });    
+    },
+  
+    play() {
+      if (this.local) {
+         this.addEventListeners(); 
+      }
+    },
+  
+    pause() {
+      if (this.local) {    
+        this.removeEventListeners();
+      }
+    },
+  
+    tick(time, delta) {
+      if (this.getMesh()?.mixer) this.getMesh().mixer.update(delta / 1000);
+    },
+    
+    update(oldData) {   
+      console.log(oldData, this.data)
+      if (oldData.visible != this.data.visible) {
+        this.el.object3D.visible = this.data.visible;     
+      }
+      
+      if (!this.local && (this.data.gesture != oldData.gesture)) {
+        this.handleGesture(this.data.gesture, oldData.gesture);
+      }
+    },
+  
+    remove() {
+      this.el.removeObject3D('mesh');
+    },
+    
+    // cache system, protection from A-Frame optimization behavior while reducing garbage collection overhead  
+    Z: 0,
+    Y: {},
+    
+    getMesh() {
+      return this.el.getObject3D(this.str.mesh)
+    },
+    
+    // local hands are controlled by one's controllers
+    // non-local hands are controlled by updates via NAF
+    local: false,
+    checkLocalHand() {
+      // this.local = this.el.components.networked.data.owner == NAF.clientId;
+      // ^while this would be more ideal, NAF doesn't populate the owner until later
+      // and we need to know this at init(), and we can't supply schema options
+      // when instantiating a template instance--so, we require adding a class for now.
+      this.el.classList.forEach(aClass => {
```

Comment:
`classList` has a `contains` api that may simplify this code to
`this.local = this.el.classList.contains("local-naf-hand")`
https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

Reply:
I tried to use that, but it wasn't available here. I think A-Frame doesn't pass it along? I'll try again and see if I had some random weirdness going on, but that was my logic here.

Reply:
Seems to work now, dunno what was happening at the moment I was coding it. Updated.